astropy/nddata/nddata.py

```diff
@@ -180,10 +178,25 @@ def __repr__(self):
 
     @property
     def data(self):
+        """
+        `~numpy.ndarray`: the data
+
+        Even though it is only enforced that the `data` is `~numpy.ndarray`
+        which does not contain objects. For use with `NDArithmeticMixin` it is
+        assumed that the data array only contains numbers.
```

Comment:
Good that you pointed that out. `NDData` only prevents the `data.dtype` to be `O` (objects). I think the most common case is that the `data` is only numbers but what if they are strings? Should I edit the `init` to prevent that? This documentation should simply highlight that `non numbers` cannot be used with arithmetic operations. Nevertheless I will move it to the class docstring of the arithmeticMixin.